spec/lib/params_sanitizer_spec.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe "ParamsSanitizer", type: :concern do
+  describe "instance methods" do
+    let(:params) { { "xml" => nil } }
+    subject { ParamsSanitizer.new(params) }
+
+    context "responds to its methods" do
+      it { expect(subject).to respond_to(:cleanse) }
+    end
+  end
+
+
+  describe "add_xml" do
+    context "when xml is nil" do
+      let(:params) { { "xml" => nil } }
+      subject { ParamsSanitizer.new(params) }
+
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is empty" do
+      let(:params) { { "xml" => "" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is not string" do
+      let(:params) { { "xml" => 123 } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is string" do
+      let(:params) { { "xml" => "<xml></xml>" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return hash" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+  end
+
```

Comment:
Can you add some more tests here for the various other things that this function is expected to do?

From reading the code, it decodes the base64 encoded `xml` and returns it, stripping the BOM and any whitespace, so it would be good to have tests for:

- Does a correctly encoded b64 string return a correct string of the contents?
- Does a badly encoded b64 string (e.g incorrect padding) return nil?
- Does the BOM get stripped correctly if present?
- Does any whitespace get stripped correctly?